pwnlib/commandline/libcdb.py

```diff
@@ -130,25 +132,41 @@
 file_parser.add_argument(
     '--unstrip',
     action = 'store_true',
-    default = False,
-    help = 'Attempt to unstrip the libc binary inplace with debug symbols from a debuginfod server'
+    dest = 'unstrip',
+    help = 'Attempt to unstrip the libc binary with debug symbols from a debuginfod server'
 )
 
-common_symbols = ['dup2', 'printf', 'puts', 'read', 'system', 'write']
+fetch_parser = libc_commands.add_parser(
+    'fetch',
+    help = 'Fetch libc database',
+    description = 'Fetch libc database'
+)
 
-def find_libc(params):
-    import requests
-    url    = "https://libc.rip/api/find"
-    result = requests.post(url, json=params, timeout=20)
-    log.debug('Request: %s', params)
-    log.debug('Result: %s', result.json())
-    if result.status_code != 200 or len(result.json()) == 0:
-        log.failure("Could not find libc for %s on libc.rip", params)
-        return []
+fetch_parser.add_argument(
+    'path',
+    nargs = '?',
+    default = context.local_libcdb,
+    help = 'Set libc-database path, If it is empty, the default path will be `context.local_libcdb` (%s)' % context.local_libcdb
+)
+
+fetch_parser.add_argument(
```

Comment:
It's weird to run `libcdb fetch` and have the command do nothing. Maybe do the update by default and ask if you want to clone/init first if the directory is still empty?

Reply:
I think init and upgrade repository is better, and remove `--init`.

Reply:
Now the second time you run `libcdb fetch` it does nothing. I'd remove all arguments and just update the database if it exists and install it first if its the first time you run the command.

Reply:
Sorry, I didn't quite understand. Currently, the expected behavior of second time run `libcdb fetch` will upgrade the repo.

![image](https://github.com/user-attachments/assets/d97b87e6-5446-4211-b5c0-8d88040f838a)

If we want update database, we should pass `-u`

![image](https://github.com/user-attachments/assets/c95705e0-10ca-40f3-9dc9-b401db3d098d)

```py
fetch_parser.add_argument(
    '-u', '--update',
    metavar = 'update',
    nargs = '*',
    choices = ['all', 'ubuntu', 'debian', 'rpm', 'centos', 'arch', 'alpine', 'kali', 'parrotsec', 'launchpad'],
    help = 'Fetch the desired libc categories'
)
```

like `libcdb fetch -u ubuntu`
![image](https://github.com/user-attachments/assets/646cf1b7-b438-47aa-b024-9f46c3aae8b1)

First time run `libcdb fetch`, it will ask if you want init repo.
![image](https://github.com/user-attachments/assets/3e88f1c0-88db-4ceb-bd22-7d6cd5bd31b4)

Reply:
Setting a default value for `-u` meet your requirements?

Reply:
Sorry, I missed the `git pull` when the libc-database repo was already cloned. This looks fine!